### DISCUSSION TITLE
Uninstalling bundle does not show Maintenance Dialog of msi

### DISCUSSION THREAD

**Discussion by hypetsch** [wixtoolset/issues#7793] [Questions]
For your deployment we have an msi package and a bundle including that msi, now we experience the following issue: 

When installing the msi and uninstalling it, everything works as expected.
But when installing with the bundle the msi does not show the "uninstall/Maintenance" dialog but instead shows the "installing dialogs" during uninstallation.

Comparing the two installation logs we saw that the msi shows the `MaintenanceWelcomeDlg` where as the `ResumeDlg` is shown when uninstalling the bundle.

![image](https://github.com/wixtoolset/issues/assets/16302578/bbe80215-741a-4553-83b0-f4ba9098d019)

We assume this is because the `ResumeDlg` has the condition [Installed AND (RESUME OR Preselected)](https://github.com/wixtoolset/wix3/blob/c5117b58502cace193732a6ba9204b68e2411355/src/ext/UIExtension/wixlib/ResumeDlg.wxs#L38) and the log shows that the `Preselected` is set to 1 in this case
`PROPERTY CHANGE: Adding Preselected property. Its value is '1'.` which does not happen when using the msi.

As we are also using the "display internal ui" option (`bal:DisplayInternalUICondition="1"`) on the `MSiPackage` it sounds similar to this ticket: #3690

Note: we recently upgraded from WiX v3 to v4 - couldn't yet verify it is related to this.

Any idea why this happens?

**Comment by robmen** [comment]
Seems like a solid hypothesis. Did you see why `Preselected` was set?

**Comment by hypetsch** [comment]
No, haven't seen it but will get a fresh log next week and try to find it out / provide it.Currently I am on vacation.

**Reply by hypetsch** [reply]
I just see that it gets set to 1 but for me it is not clear why. Attached I added the logs of an uninstallation until the point where the msi shows the dialog with the "Install" button. The MSiPackage element in the bundle looks like this:
```
<MsiPackage SourceFile="$(var.Tcs.Samwin.Installer.Msi.TargetPath)" Compressed="yes" EnableFeatureSelection="yes" bal:DisplayInternalUICondition="1">
   <MsiProperty Name="SAMWIN_MODULES" Value="[SAMWIN_MODULES]" />
   <MsiProperty Name="INSTALLDIR" Value="[INSTALLDIR]" />
</MsiPackage>
```

[installer.log](https://github.com/wixtoolset/issues/files/13202350/installer.log)
[installer_000_Tcs.Samwin.Installer.msi.log](https://github.com/wixtoolset/issues/files/13202351/installer_000_Tcs.Samwin.Installer.msi.log)

**Reply by robmen** [reply]
I'm 99.9999% certain this is why: `EnableFeatureSelection="yes"`. You've said your BA will take over feature selection, which means `Preselected` will be set.

**Reply by hypetsch** [reply]
Setting EnableFeatureSelection to "no" nor removing it didn't help. Btw. even after setting it to "no" selecting features in the msi was possible, so think I didn't understand what this settings is for :).

 As it would be ok for us to hide the msi UI during uninstallation (it is a "click next/next" anyways), I played around with the "DisplayInternalUICondition" but couldn't find a solution to entirely hide the msi UI during uninstallation.

**Reply by hypetsch** [reply]
Think we found a workaround for the original issue by hiding the UI during uninstallation.

We now have the following code in place checking for the value of `WixBundleAction` to determine whether the product is being installed or uninstalled.

```xml
<MsiPackage SourceFile="$(var.Tcs.Samwin.Installer.Msi.TargetPath)" Compressed="yes" bal:DisplayInternalUICondition="WixBundleAction&lt;&gt;4">
   <MsiProperty Name="SAMWIN_MODULES" Value="[SAMWIN_MODULES]" />
   <MsiProperty Name="INSTALLDIR" Value="[INSTALLDIR]" />
</MsiPackage>
```

First tests seem to be ok.

Maybe documenting the possible values of `WixBundleAction` / [BOOTSTRAPPER_ACTION](https://github.com/wixtoolset/wix/blob/b84202794f5fd4f21e0fb1fc74c2f5d531035db3/src/api/burn/WixToolset.BootstrapperCore.Native/inc/BootstrapperEngine.h#L18) in https://wixtoolset.org/docs/tools/burn/builtin-variables/ for would be nice.
Btw. personally I like explicitly specifying the enum values for such "public enums" so they do not get erroneously changed if a new value is added in between.
